rllib/policy/policy.py

```diff
@@ -362,6 +370,377 @@ def __init__(
+            [
+                self._lazy_tensor_dict(step_out.data.sample_batch)
+                for step_out in processed
+            ]
+
+        action_connector_input_data = [
+            self.compute_actions_from_input_dict(
+                step_out.data.sample_batch,
+                explore=explore,
+                # This may be inaccurate when processing large batches
+                kwargs=kwargs,
+                **kwargs,
+            )
+            for step_out in processed
+        ]
+
+        # Turn action lists from ES and ARS into arrays
+        action_connector_input_data = [
+            (np.array(a), s, i) if isinstance(a, list) else (a, s, i)
+            for a, s, i in action_connector_input_data
+        ]
+
+        actions = list()  # We return [BATCH_DIM, ACTION_DIM, ...]
+        rnn_states = list()  # We return [BATCH_DIM, STATE_DIM, ...]
+        fetches = defaultdict(list)  # We return a dict shaped similarly as
+        # {"f1": [BATCH_SIZE, ...], "f2": [BATCH_SIZE, ...]}.
+
+        for output_data, env_id, agent_id, agent_idx in zip(
+            action_connector_input_data, env_ids, agent_ids, range(len(env_ids))
+        ):  # All
+            # lengths of batched inputs assumed to be equal here, choice of env_ids is
+            # arbitrary
+
+            # Construct the "input_dict" manually here, since we have one input dict
+            # per agent
+            input_dict = {
+                SampleBatch.NEXT_OBS: next_obs_batch,
+            }
+            if reward_batch is not None:
+                input_dict[SampleBatch.REWARDS] = reward_batch[agent_idx]
+            if dones_batch is not None:
+                input_dict[SampleBatch.DONES] = dones_batch[agent_idx]
+            if info_batch is not None:
+                input_dict[SampleBatch.INFOS] = info_batch[agent_idx]
+            if t_batch is not None:
+                input_dict[SampleBatch.T] = t_batch[agent_idx]
+
+            #  Post-process policy output by running them through action connectors.
+            ac_data = ActionConnectorDataType(
+                env_id="0",
+                agent_id="0",
+                input_dict=input_dict,
+                output=output_data,
+            )
+
+            self.agent_connectors.on_policy_output(ac_data)
+
+            _action, _rnn_state, _fetches = self.action_connectors(ac_data).output
+            actions.extend(_action)
+            rnn_states.append(np.array(_rnn_state))
+            for key, item in _fetches.items():
+                fetches[key].extend(item)
+
+        # Turn fetches into a dict and conform all batches inside into TensorType
+        fetches = dict(fetches)
+        for key, value in fetches.items():
+            fetches[key] = np.array(value)
+
+        return (
+            np.array(actions),
+            convert_to_numpy(rnn_states),
+            convert_to_numpy(fetches),
+        )
+
+    @property
+    def action_connectors_created(self):
```

Comment:
are these properties suppose to be public APIs? also docstring?

Reply:
should we just make them either super private (with two underscores in front) ? or just move them entirely into the logic of compute_actions_from_inputs() method. I don't see any re-use just yet.

Reply:
Inlined! Thanks